### PR TITLE
fix(network): Try reparsing the socket address

### PR DESCRIPTION
**Pull request by yejiayu**
# Pull Request Template

### Requirements
ref #250 

When parsing the socket address, it will retry if it fails.

Note: There is still a problem updating the `network` configuration

@jiangxianliang007 Please test this branch on the k8s cluster.

### CODE REVIEW

**Comment by yejiayu** [comment]
PTAL @kaikai1024 @leeyr338

**Comment by yejiayu** [comment]
Is it normal for ci to fail?

**Comment by u2** [comment]
> Is it normal for ci to fail?

No. Check format.

**Comment by ouwenkg** [comment]
Checking clippy failed.

**Comment by yangby-cryptape** [comment]
> error[E0308]: mismatched types
>    --> cita-network/src/connection.rs:226:41

Have you built the project before pulling request? This is not a lint error!

CI is really important for this situation.

**Comment by yangby-cryptape** [comment]
> Is it normal for ci to fail?

Have you checked the CI logs before you asked?

It is OK if you just confused with the CI logs, ask and someone will answer it, but, if you just are lazy ...
Come on, u are a grown-up!

**Comment by yejiayu** [comment]
> Have you checked the CI logs before you asked?
> 
> It is OK if you just confused with the CI logs, ask and someone will answer it, but, if you just are lazy ...
> Come on, u are a grown-up!

I checked the previous pr and found that many ci failed, so I raised this question.

**Comment by kaikai1024** [comment]
> > Have you checked the CI logs before you asked?
> > It is OK if you just confused with the CI logs, ask and someone will answer it, but, if you just are lazy ...
> > Come on, u are a grown-up!
> 
> I checked the previous PR and found that many ci failed, so I raised this question.

Because the circleCI was not ready that time.

File: cita-network/src/connection.rs [hunk 1]
```diff
@@ -427,7 +429,7 @@ pub fn manage_connect(config_path: &str, rx: Receiver<DebouncedEvent>, task_send
 
 fn config_peer(config: &[PeerConfig]) -> Vec<(u32, SocketAddr, String)> {
```
Comment (kaikai1024) [inline cita-network/src/connection.rs]: Use `init_config_peer` instead of `config_peer`? Then why change this function?

Comment (yejiayu) [reply cita-network/src/connection.rs]: The update configuration also depends on "config_peer".

**Comment by jiangxianliang007** [comment]
Tested through the K8S cluster

**Comment by kaikai1024** [comment]
> Tested through the K8S cluster

We'll no release a patch version, That's right?

**Pull request merged**